src/libsyntax/json.rs

```diff
@@ -283,13 +283,14 @@ impl DiagnosticSpanLine {
     fn line_from_filemap(fm: &syntax_pos::FileMap,
                          index: usize,
                          h_start: usize,
-                         h_end: usize)
-                         -> DiagnosticSpanLine {
-        DiagnosticSpanLine {
-            text: fm.get_line(index).unwrap().to_owned(),
-            highlight_start: h_start,
-            highlight_end: h_end,
-        }
+                         h_end: usize) -> Option<DiagnosticSpanLine> {
+        fm.get_line(index).map(|text| {
```

Comment:
Add a comment here saying what could cause this to return `None` (as the commit message does).